src/codec.rs

```diff
@@ -1207,13 +1258,20 @@ macro_rules! impl_codec_through_iterator {
 			fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
 				<Compact<u32>>::decode(input).and_then(move |Compact(len)| {
 					input.descend_ref()?;
-					let result = Result::from_iter((0..len).map(|_| Decode::decode(input)));
+					let result = Result::from_iter((0..len).map(|_| {
```

Comment:
Most of the time containers are implemented to always increase the capacity by two when hitting the limit. This means here that we may only decode 190 elements while the capacity is already at 256. What I want to say is that you will not track the size correctly.

Reply:
True. Initially I wanted to avoid the complexity of keeping track of this. I saw that other libraries did the same. For example the `heapsize` lib from `servo`: https://github.com/servo/heapsize/blob/master/src/lib.rs#L295 . But anyway, done. PTAL !